### ISSUE TITLE
LACP created in a switch assigned to virtual chassis is not available for the second switch of statck

### ISSUE THREAD

**Issue by jcralbino** [netbox-community/netbox#6531]
### NetBox version

v2.11.4

### Python version

3.8

### Steps to Reproduce

Using the GUI we performed this:

1. Created a virtual chassis: stack-1-withoutmaster
2. Create first switch: stack1-sw1
3. Create second switch: stack1-sw2
4. Associate them to the stack-1-withoutmaster, without selecting a master.
       As i do not associate a master both switches have only their own interfaces

5. Create Link Aggregation type (lag-stack1-port1) in stack1-sw1
6. Associate lag-stack-port1 to stack1-sw1 interface e1/1
7. Failed to associate lag-stack-port1 on stack1-sw2.

This is not possible now in 2.11.4 , It was possible in 2.11.2

### Expected Behavior

1. Created a virtual chassis: stack-1-withoutmaster
2. Create first switch: stack1-sw1
3. Create second switch: stack1-sw2
4. Associate them to the stack-1-withoutmaster, without selecting a master.
       As i do not associate a master both switches have only their own interfaces

5. Create Link Aggregation type (lag-stack1-port1) in stack1-sw1
6. Associate lag-stack-port1 to stack1-sw1 interface e1/1
7.  Associate lag-stack-port1 on stack1-sw2 on interface e1/1

### Observed Behavior

LAG object not provided in the dropdown menu

**Comment by jcralbino** [comment]
@candlerb i have opened this bug

**Comment by jeremystretch** [comment]
IMO this is desired behavior. The LAG selection widget relies on the `device_id` filter to filter the list of interface choices returned. This filter restricts the interfaces queryset to only those assigned to the specified device or, _if the device is a VC master_, to a peer VC member. Altering this behavior would break the filter for other uses.

> As i do not associate a master both switches have only their own interfaces

Setting a VC master and defining the LAG interfaces on that master will resolve the issue.

**Comment by jcralbino** [comment]
This is a undesired behaviour in my opinion as it will limit the utilisation of the virtualchassis.

There many use cases where setting a master is not mandatory, as within the multi chassis environment were we do not have a shared control plane. Therefore the interfaces will be kept in each member and not renamed in those cases 

This is something I tested in previous versions and that was working

**Comment by jeremystretch** [comment]
> as within the multi chassis environment were we do not have a shared control plane

Ah, I think this is the crux of your problem. A virtual chassis is intended to model a shared control plane. If the devices have independent control planes (as in a multi-chassis LAG deployment), a virtual chassis should not be used, because they are still independent devices. Each device in a multi-chassis LAG setup has its own LAG interface that needs to be modeled discretely.

**Comment by jcralbino** [comment]
What I have seen is that we are not enforcing here the placement of a master. I see here two possibilities 
Virtual chassis with master configured : 
- then we have a stack of switches where the control plane is handled by the master and therefore a rename of interfaces of the second member is mandatory as the master will own this interfaces 
 - typical use case for switches 

Virtual chassis without master:
 - the devices are part of a chassis or cluster. Maintaining its management plane
 - several possible use case for 
Example
   1) cluster of firewall, where active / passive is used but interfaces are owned by each device. Each interface should be configured exactly the same 
   2) cluster of load balancer, with similar approach for the firewall
   3) may be used to support all types of multi chassis link aggregation. Like a 
VPC domains in Cisco, where a  link aggregation can be used across two devices that do no have a master device . ( current limited on the way lag interfaces work )
 

If these use cases are not be supported then the utilisation of a master should be enforced.

But as this was working in 2.11.2, and probably other versions although i can't confirm , I believe that enforcing the use of lag only when a master is configured is too limited . Is there a strong technical reason for this, as it seems the database is already prepared to support this option.

**Comment by jcralbino** [comment]
_The LAG selection widget relies on the device_id filter to filter the list of interface choices returned. This filter restricts the interfaces queryset to only those assigned to the specified device or, if the device is a VC master, to a peer VC member. Altering this behavior would break the filter for other uses._

Can we have a filter that restricts the interfaces queryset to only those assigned to the specified device or, if the device is part of a VC member, to the other peer VC member?

**Comment by jcralbino** [comment]
> _The LAG selection widget relies on the device_id filter to filter the list of interface choices returned. This filter restricts the interfaces queryset to only those assigned to the specified device or, if the device is a VC master, to a peer VC member. Altering this behavior would break the filter for other uses._
> 

Can we change the filter in order to restrict  the interfaces queryset to only those assigned to the specified device or, if the device is a VC member, to the other peer VC member.

**Comment by jeremystretch** [comment]
No, I'm afraid we cannot. I explained above that this is an intentional design choice. You'll need to define a LAG interface for each member of a multi-chassis LAG.

**Comment by jcralbino** [comment]
Hello Jeremy. Thanks for the reply.

Then we should add to the documentation that a master in a virtual chassis is mandatory. As it is not clear to me what is the use case where it makes sense to have a virtual chassis without a master .

I am sure a lot of people are using this without a master, but the documentation should make clear what is the use case for when a virtual chassis without master can be used .
